homeassistant/components/zwave_js/config_flow.py

```diff
@@ -410,18 +410,17 @@ async def async_step_manual(
             _LOGGER.exception("Unexpected exception")
             errors["base"] = "unknown"
         else:
-            await self.async_set_unique_id(
-                str(version_info.home_id), raise_on_progress=False
-            )
-            # Make sure we disable any add-on handling
-            # if the controller is reconfigured in a manual step.
-            self._abort_if_unique_id_configured(
-                updates={
-                    **user_input,
-                    CONF_USE_ADDON: False,
-                    CONF_INTEGRATION_CREATED_ADDON: False,
-                }
-            )
+            for home_id in (version_info.home_id, str(version_info.home_id)):
+                await self.async_set_unique_id(home_id, raise_on_progress=False)
```

Comment:
It's weird to set a unique ID twice. Why not just normalize it before and then do business as usual?



```python
        for entry in self._async_current_entries(include_ignore=True):
            if isinstance(entry.unique_id, int):
                self.hass.config_entries.async_update_entry(entry, unique_id=str(entry.unique_id))

```

Reply:
I guess you'd could import and call `homeassistant.components.zwave_js.async_migrate_entry(hass, entry)` for each entry as well, to set of the same migration as the integration setup does.

Reply:
It's not clear what normalizing means here as there are two things to compare.

Yes, I think we could make sure that all entries are migrated here before setting the unique id.

Reply:
With normalizing I mean we make all unique IDs strings.

Reply:
As in on startup we migrate all config entries to use a string? Or we do a string comparison between unique IDs in `_abort_if_unique_id_configured`? Or I guess we could do both. On load convert the config entries unique ID to a string. In comparison checks convert the input unique ID as a string?

Reply:
We can call it when something is discovered and during async_setup.

Reply:
I tried to do this in async_setup because I thought it would be cleaner but it would have required significant changes to tests. We can move it later if desired, for now we do it during the config flow

Reply:
nvm, I figured it out